tensorflow/core/kernels/random_op_gpu.cu.cc

```diff
@@ -141,7 +141,7 @@ struct FillPhiloxRandomKernel<Distribution, false> {
       const typename Distribution::ResultType samples = dist(&gen);
       copier(&data[offset], samples);
 
-      offset += (total_thread_count - 1) * kGroupSize;
+      offset += total_thread_count * kGroupSize;
       gen.Skip(total_thread_count - 1);
```

Comment:
I thought so too, but when I changed this, it gave different results than the CPU version of the initializer. Somehow, the generator does not generate repeated values even though this code appears to only skip `total_thread_count - 1` indices in the randomized sequence (i.e. seems like we should expect that the 0th and `total_thread_count - 1`st threads would produce the same results in successive steps). Perhaps there is an off-by-one issue inside the generator code that is consistent between CPU and GPU versions?

Do you have thoughts/recommendations for how to proceed here?